.buildkite/pipeline.yml

```diff
@@ -0,0 +1,11 @@
+env:
+  IMAGE_ID: xcode-12.5.1
+agents:
+  queue: 'mac'
```

Comment:
@jkmassel @AliSoftware TIL that Buildkite offers a native way to specify [default `env` and `agents`](https://buildkite.com/docs/pipelines/defining-steps#step-defaults). I think we should adopt this approach in our other pipelines instead of the custom anchor for the `env` value.

I haven't tested this, but I expect that if we define a default `env` with a `FOO` variable and then define a step-specific `env` with a `BAR` variable, the step will have access to both `FOO` and `BAR`.

Reply:
I like the pipeline level default and agree it makes sense to use it.

But unlike you my guess would be that any step-specific env would completely override the default. Otherwise there wouldn't be a way to undefine FOO in some steps… so in any case definitively worth checking the behavior in such a case.

Reply:
@mokagio Huh, actually this might not behave like either of us expect, as according to the BuildKite documentation, the `env` set at the top level… always takes precedence over the ones set on each `step`?!

> Top level pipeline environment variables will override what is set in the env attribute of an individual step.

[Source: BuildKite docs](https://buildkite.com/docs/pipelines/environment-variables#environment-variable-precedence-job-environment)

Which is a strange choice from them imho, and iiuc means that we wouldn't be able to provide e.g. a different `IMAGE_ID` just for a specific step while using the top-level-defined one as a default, because the top-level one would always override them all… 😒